src/main/java/core/basesyntax/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,15 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.ReportGenerator;
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    @Override
+    public String generateReport(Map<String, Integer> inventory) {
+        StringBuilder report = new StringBuilder("fruit,quantity\n");
+        inventory.forEach((fruit, quantity) ->
+                report.append(fruit).append(",").append(quantity).append("\n")
+        );
+        return report.toString();
+    }
```

Comment:
Make sure to follow the encapsulation principle. The method 'generateReport' should be as closed as possible. Consider declaring 'report' as a private method field if it's going to be used in other methods of this class in the future, or keep it local to the method if it's not going to be reused.